conans/client/conf/config_installer.py

```diff
@@ -1,12 +1,40 @@
 import os
-from conans.tools import unzip
 import shutil
+from six import string_types
+
+try:
+    # urlparse was moved to urllib in python 3
+    from urllib.parse import urlparse
+except ImportError:
+    # location for python 2
+    from urlparse import urlparse
+
+from conans.tools import unzip
 from conans.util.files import rmdir, mkdir
 from conans.client.remote_registry import RemoteRegistry
 from conans import tools
 from conans.errors import ConanException
 
 
+def _remove_credentials(resource):
+    """
+    Remove credentials from resource url/file path
+
+    :param resource: string with url or file path
+    :return: resource with credentials removed if present
+    """
+    if not isinstance(resource, string_types):
```

Comment:
This comparison doesn't feel very pythonic.

Reply:
Actually I think I would completely remove it. This is a pre-condition for this method, if this is not a string, it will cause issues elsewhere in the code. Getting here with ``resource`` not being a string should be considered a bug, and this only masks such bug.
Don't check types unless there is strong evidence that it is necessary.

Reply:
@lasote That comparison is using the [string_types](https://stackoverflow.com/questions/11301138/how-to-check-if-variable-is-string-with-python-2-and-3-compatibility) from six module.

@memsharded I'm ok removing this check, going to replace with:

```
if not resource:
    return resource
```